### PR TITLE
Application Portal Shell

### LINKED ISSUE #75: Application Portal Multi-Step Form Skeleton
* Create a new page called `/apply`
* Get the query param step using the `useRouter` hook which will hold the step of the application form.
   * https://nextjs.org/docs/pages/api-reference/functions/use-router
   * https://www.geeksforgeeks.org/how-to-get-query-parameters-from-url-in-next-js/
* Conditionally render a different step based off of the `step` query param
   * Each step should just render the title of the step.
* Add a navbar on the bottom of the form with links to each step

** See last year's code as an example, but please follow the above approach for rendering steps. We just want the portal to be one next.js page rather than one page per step.

## Steps

1. About You `/apply?step=about-you`
2. Your Story `/apply?step=your-story`
3. Profile Links `/apply?step=profile-links`
4. Agreements `/apply?step=agreements`
5. Optional Questions `/apply?step=optional-questions`
6. Application Review `/apply?step=application-review`

### PR DESCRIPTION
**Pull request by basokant**
closes #75 

# Quick Overview of Changes

- add `/apply` route with routing using query params for each step in the form
- created stub components for each step in the form
- followed styling for navbar, footer menu, and 
- added constant `applySteps` for source as source of truth for steps in the application
- added `dev` branch to trigger on GH actions

# Checklist

- [x] If any frontend changes were made, include a screenshot/demo in the overview
- [x] Checked all uses of changed component(s)/function(s)
- [x] If any changes were made to our backend services, at least one (happy path) unit test was added OR a `TODO` comment was added to create one.
- [x] Check that CI is passing.
- [x] If no (re-)reviews after 1-2 days, ping the web leads on the discord to remind them to review the PR.

@basokant @ArsalaanAli

### CODE REVIEW

File: src/components/apply/navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+};
+
+function getPreviousStep(stepIndex: number | null): ApplyStep | null {
+  if (stepIndex === null) return null;
+  return applySteps[stepIndex - 1]?.step ?? null;
+}
+
+function getNextStep(stepIndex: number | null): ApplyStep | null {
+  if (stepIndex === null) return null;
+  console.log(applySteps[stepIndex], applySteps[stepIndex + 1]);
```
Comment (hunterchen7) [inline src/components/apply/navigation.tsx:17]: please remove this :-1:

**Review by hunterchen7 — DISMISSED** [review]
remove the console log

File: src/components/apply/navigation.tsx (hunk 1 above)
Comment (basokant) [reply src/components/apply/navigation.tsx]: removed

File: src/components/apply/form/index.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      return <ApplicationForm />;
+    case "links":
+      return <LinksForm />;
+    case "agreements":
+      return <AgreementsForm />;
+    case "optional":
+      return <OptionalForm />;
+    case "review":
+      return <ReviewForm />;
+  }
```
Comment (LordExodius) [inline src/components/apply/form/index.tsx]: no default case for `null` step 😞

**Review by LordExodius — APPROVED** [review]
lgtm (👎 because he fixed it too fast for me to leave a review)

**Pull request merged**